portfolio/src/main/webapp/script.js

```diff
@@ -32,16 +33,71 @@ function getRandomMessageUsingArrowFunctions() {
   });
 }
 function getjson() {
-  fetch('/data').then(response => response.json()).then((data) => {
+    console.log("test");
+    fetch('/loginstatus').then(response => response.json()).then((loginstatus) => {
+        const email = loginstatus.email;
+        const redirectUrl = loginstatus.redirectUrl;
+        const dataElement = document.getElementById('data-container');
+
+
+        // if not logged in
+        if(email == ""){
+            dataElement.innerText = "Log in to see comments!";
+            console.log("test1");
+        }
+        else {
+            fetch('/data').then(response => response.json()).then((data) => {    
+                dataElement.innerText = '';
+                for (var i =0; i < data.length; i++){
+                    if (data[i].text != ""){
+                    dataElement.appendChild(createListElement(data[i].email + ' : '+ data[i].text));
+                    console.log(data[i].text);
+                    }
+                }
+
+                const createform = document.createElement('form');
+                createform.setAttribute("action", "/data");
+                createform.setAttribute("method", "POST");
     
-    const dataElement = document.getElementById('data-container');
-    dataElement.innerText = '';
 
-    for (var i =0; i < data.length; i++){
-        dataElement.appendChild(createListElement('Comment ' + [i+1] +' : '+ data[i].text));
-        console.log(data[i].text);
-    }
-  });
+                const textarea = document.createElement("textarea");
+                const msg = document.createTextNode("Add a comment here");
+                textarea.appendChild(msg);
+                textarea.name = "text-input";
+                createform.appendChild(textarea);
```

Comment:
Great job researching how to create this form from Javascript! I tried this on your live server, and it works great!